File: core/src/main/java/com/mcreater/amclcore/command/ansi/DefaultTheme.java
package com.mcreater.amclcore.command.ansi;

import org.fusesource.jansi.Ansi;

public class DefaultTheme implements ColorTheme {
    public Ansi.Consumer applyFatal() {
        return a -> a.fg(Ansi.Color.RED);
    }

    public Ansi.Consumer applyError() {
        return a -> a.fg(Ansi.Color.RED);
    }

    public Ansi.Consumer applyWarning() {
        return a -> a.fg(Ansi.Color.YELLOW);
    }

    public Ansi.Consumer applyInfo() {
        return a -> a.fg(Ansi.Color.GREEN);
    }

    public Ansi.Consumer applyDebug() {
        return a -> a.fg(Ansi.Color.CYAN);
    }

    public Ansi.Consumer applyTrace() {
        return a -> a.fg(Ansi.Color.BLUE);
    }
}

File: core/src/main/java/com/mcreater/amclcore/command/ansi/DefaultTheme.kt
package com.mcreater.amclcore.command.ansi

import org.fusesource.jansi.Ansi


class DefaultTheme : ColorTheme {
    override fun applyFatal(): Ansi.Consumer {
        return Ansi.Consumer { it.fg(Ansi.Color.RED) }
    }

    override fun applyError(): Ansi.Consumer {
        return Ansi.Consumer { it.fg(Ansi.Color.RED) }
    }

    override fun applyWarning(): Ansi.Consumer {
        return Ansi.Consumer { it.fg(Ansi.Color.YELLOW) }
    }

    override fun applyInfo(): Ansi.Consumer {
        return Ansi.Consumer { it.fg(Ansi.Color.GREEN) }
    }

    override fun applyDebug(): Ansi.Consumer {
        return Ansi.Consumer { it.fg(Ansi.Color.CYAN) }
    }

    override fun applyTrace(): Ansi.Consumer? {
        return Ansi.Consumer { it.fg(Ansi.Color.BLUE) }
    }
}

